src/AutomaticVersionRanges.targets

```diff
@@ -0,0 +1,42 @@
+<Project>
+
+  <PropertyGroup>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == '' And '$(Configuration)' == 'Debug'">false</AutomaticVersionRangesEnabled>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == '' And '$(IsPackable)' == 'false'">false</AutomaticVersionRangesEnabled>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == '' And '$(ManagePackageVersionsCentrally)' == 'true'">false</AutomaticVersionRangesEnabled>
+    <AutomaticVersionRangesEnabled Condition="'$(AutomaticVersionRangesEnabled)' == ''">true</AutomaticVersionRangesEnabled>
+  </PropertyGroup>
+
+ <UsingTask TaskName="ConvertToVersionRange" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+  <Task>
+    <Code Source="$(MSBuildThisFileDirectory)ConvertToVersionRange.cs" />
+  </Task>
+  </UsingTask>
+
+<Target Name="ConvertProjectReferenceVersionsToVersionRanges" AfterTargets="_GetProjectReferenceVersions" Condition="'$(AutomaticVersionRangesEnabled)' == 'true'">
```

Comment:
Since it's touching private targets, have you tried this with the net9 SDK, since we're only a few weeks from that?

Reply:
The 8.0.400 SDK's files already have the 9.X SDK version check in them to alter the audit mode default, so seems unlikely for that to change.

I'll double check to makes sure though.

Reply:
Oh, you're talking about the project reference target again. That's been in Particular.Packaging for years now, spanning multiple .NET releases. It doesn't have anything to do with the NuGet audit changes, so there's no reason to expect it to change/break with .NET 9.